### ISSUE TITLE
Incorrect earnings (because of lack of info on pension contributions)

### ISSUE THREAD

**Issue by MattHJensen** [open-source-economics/Tax-Calculator#1549]
A researcher at the University of Pennsylvania asked: 

> I'd like to simulate changes in personal tax liability under different income source assumptions (e.g. W-2 wages vs. self-employment vs. IRA distributions).
> 
> Example comparing two family scenarios:
> 
> If a family earning $50k in W-2 wages contributes $10k to an employer-sponsored retirement plan, their "earned income" for EITC purposes is $40k. However, if the same family earns $50k in self-employment income and contributes $10k to a traditional IRA (above the line deduction), the EITC worksheets do not, in fact, account for the contributions and their earned income for EITC purposes remains $50k. In effect, the code penalizes the self-employed w.r.t. the EITC. In these two cases, ACA credits would also play a big role in overall tax liability, if we assume the family must purchase private insurance in the self-employed scenario.
> 
> [How do I do this]

**Comment by feenberg** [comment]
On Thu, 14 Sep 2017, Matt Jensen wrote:

> 
> A researcher at the University of Pennsylvania asked:
>
>       I'd like to simulate changes in personal tax liability under
>       different income source assumptions (e.g. W-2 wages vs.
>       self-employment vs. IRA distributions).
>
>       Example comparing two family scenarios:
>
>       If a family earning $50k in W-2 wages contributes $10k to an
>       employer-sponsored retirement plan, their "earned income" for
>       EITC purposes is $40k. However, if the same family earns $50k in

Both earnings and AGI are reported as 40K

>       self-employment income and contributes $10k to a traditional IRA
>       (above the line deduction), the EITC worksheets do not, in fact,
>       account for the contributions and their earned income for EITC
>       purposes remains $50k. In effect, the code penalizes the

Earnings is 50K and AGI is still 40K. At lower incomes this the EITC would 
be calculated from earnings, and parity with the employed would be 
maintained. At this income level, the SE lose relative to the employed.

>       self-employed w.r.t. the EITC. In these two cases, ACA credits
>       would also play a big role in overall tax liability, if we
>       assume the family must purchase private insurance in the
>       self-employed scenario.
>
>       [How do I do this]

I don't think could be done with the online calculator. In the source code 
it would be a matter of moving the IRA adjustment into wages. This is most 
easily done as a modification to the data before the calculator does the 
tax calculation. Did we make any progress on allowing generic calculations 
in TB? This is a good example of something that could be done that way.

dan

**Comment by MattHJensen** [comment]
Thanks @feenberg. I reported back to the user that he could use the [tc CLI](http://open-source-economics.github.io/Tax-Calculator/#cli) to accomplish this work and pointed him towards your suggestion to modify the data. 

This cannot be done on TaxBrain right now. 

Closing this issue. If further questions come up, I'll reopen it.

**Comment by martinholmer** [comment]
The question posed in issue #1549 focuses on analyzing tax liabilities for two individuals (one an employee and the other self-employed) with the same earnings and the same contribution to a tax-preferred retirement plan.  The employee makes a contribution to an employer-sponsored plan (like a 401k plan) and the self-employed individual makes a contribution to a traditional IRA.

Given my understanding of the way the tax laws work (which could be incorrect), this question reveals a flaw in both Tax-Calculator data and Tax-Calculator logic.  All of what I say below is based on this understanding, so it is important for others on the development team to check whether my understanding is correct or not.

Given my understanding, here is my explanation of the flaws.

First some definitions: **gross earnings** are total earnings, **employee contributions** are contributions to an employer-sponsored tax-preferred retirement plan, and **net earnings** are gross earnings minus employee contributions.

My understanding is that an employee's W-2 box 1 and Form 1040 line 7 contain **net earnings**, which is included in AGI.  So, the information from the IRS-SOI PUF is adequate to calculate income tax liability.

But my understanding is that an employee's FICA tax is calculated using **gross earnings**.  That means that individuals with a positive **employee contribution** will have their payroll tax under-estimated in Tax-Calculator.

I have no idea what the CPS data (that are included in `puf.csv` and in `cps.csv`) contain for the earnings variables (`e00200`, etc.).  Is it **gross earnings** or **net earnings**?  I can imagine it is net because they may ask the CPS interviewee what's on their W-2 form, but I really don't know.  If the CPS data contain **gross earnings**, then Tax-Calculator is over-estimating income tax liability for individuals with positive **employee contributions**.   If the the CPS data contain **net earnings**, then Tax-Calculator is under-estimating payroll tax liability for individuals with positive **employee contributions**.

What's missing from the Tax-Calculator data files is **employee contributions**.  And the missing Tax-Calculator logic needs to add **employee contributions** to **net earnings** to get the **gross earnings** needed for payroll tax calculation.  These are not small amount that are missing.  Each year, tens of millions of individuals have positive **employee contributions** that are typically several thousands of dollars.

Does this make sense?  Or am I confused about how the tax laws and tax reporting work?

@MattHJensen @feenberg @Amy-Xu @andersonfrailey @hdoupe @GoFroggyRun

**Comment by martinholmer** [comment]
Issue #1156 --- Tax-Calculator showing lower wages and salaries than expected --- discussed reasons for this, but did not consider the point raised in issue #1549: that Tax-Calculator wages and salaries are **net earnings** (that is, **gross earnings** minus **employee contributions** to employer-sponsored tax-preferred retirement plans).  

Do the wages and salaries totals being used in the #1156 comparisons include **employee contributions** or not?  If they do, then this is an obvious reason why the Tax-Calculator wages and salaries, which do not include **employee contributions**, are lower.

@MattHJensen @feenberg @Amy-Xu @andersonfrailey @hdoupe @GoFroggyRun

**Comment by andersonfrailey** [comment]
@martinholmer, I'm looking at the CPS documentation and the wages and salaries variable, `wsal-val`, is described as "total wage and salary earnings."

The exact wording of the question is:
>How much did (name/you) earn from this employer before taxes and other deductions during [YEAR]?

And then in the body of the text you see:
> Also, money income does not reflect...full or partial contributions for retirement programs, medical and educational expenses, etc.

([page 9-3](http://www.nber.org/cps/cpsmar2015.pdf))

Given this, it appears to me that the CPS data contains gross earnings.

Unfortunately, I haven't seen any information in the CPS regarding how much an individual contributes to an employer-sponsored tax-preferred retirement plans.

**Comment by martinholmer** [comment]
@andersonfrailey,  Thanks for your research into the earnings variables in the CPS data, which confirms that CPS earnings are what I call **gross earnings** in issue #1549.

**Comment by martinholmer** [comment]
It has been a week and nobody has corrected [my argument](https://github.com/open-source-economics/Tax-Calculator/issues/1549#issuecomment-329816059) that earnings in the PUF data are substantially understated for payroll tax calculation.  Is the lack of comment an implicit agreement?  Can anyone confirm that payroll taxes are levied on **gross earnings**?  By gross earnings, I mean earnings before contribution to employer-sponsored tax-favored retirement plans (like a 401k plan) are subtracted and recorded on the W-2 and 1040 forms are net earnings.

If, in fact, payroll taxes are levied on gross earnings, we have a substantial data problem with the `puf.csv`.

@MattHJensen @feenberg @Amy-Xu @andersonfrailey @hdoupe @GoFroggyRun

**Comment by MattHJensen** [comment]
@martinholmer, I plan to look into this closely today or tomorrow.

@andersonfrailey, thanks very much for looking into the CPS side of this. I'd be very interested in your view of the PUF side things as well.

**Comment by martinholmer** [comment]
@MattHJensen said in issue #1549:
> I plan to look into this closely today or tomorrow.
>
> @andersonfrailey, thanks very much for looking into the CPS side of this. I'd be very interested in your view of the PUF side things as well.

One final thought before you plunge into this issue.  If my thinking is correct, when using PUF records (in `puf.csv`) we have correct earnings for income tax calculation and under-estimated earnings for payroll tax calculation.  On the other hand, if my thinking is correct, when using CPS records (in either`puf.csv` or `cps.csv`) we have over-estimated earnings for income tax calculation and correct earnings for payroll tax calculation.

**Comment by MattHJensen** [comment]
@martinholmer, my reading leads me to confirm your view of this issue regarding PUF variables: non-roth tax-deductible retirement contributions are missing from our payroll tax calculations. 

I have not yet delved into the CPS documentation. 

I don't have a firm view yet on how best to proceed, but I will offer a link to the Tax Policy Center's 2005 documentation that contains an extensive discussion of how they impute these amounts. See Appendix A for a full discussion. 

I must say, though, that there is a very confusing aspect in TPC's documentation. See the the first full paragraph on page 32, which first mentions Appendix A. This paragraph claims that TPC calculates payroll taxes before imputing retirement contributions, but I don't see how that could be possible given our understanding of the problem. If someone confirms my confusion, I will write to TPC modelers to ask them about this.

**Comment by martinholmer** [comment]
@MattHJensen,  Thanks for the TPC link in issue #1549.  I'll read that later in the week.

**Comment by martinholmer** [comment]
On page 32 of the TPC documentation, it says:
> After the parameter file is input, the model reads in a single record from the PUF-CPS
matched database and ages the record. The first-stage growth factors are applied to all the
appropriate fields and the Stage II-adjusted weight is read in from the database. Following this,
federal payroll taxes for Social Security and Medicare are calculated. After this, the steps and
order of calculations generally follows Form 1040. AGI is determined first and then the standard
deduction to which the record is entitled as well as any itemized deductions are calculated. When
calculating AGI, the deduction from income for contributions to Individual Retirement
Arrangements (IRAs) is determined by the model’s retirement savings module. In addition to
determining traditional IRA contributions, the retirement savings module determines
contributions to Roth IRAs and defined contribution pension plans. Details can be found in
appendix A. 

@MattHJensen, Could this be saying that the earnings for PUF records in the PUF-CPS file **already** include tax-favored employee pension contributions and are, therefore, what I've been calling **gross earnings**?  If that were the case, then the payroll tax calculations would be correct, but there would be a need to get **net earnings** for the income tax calculations.  That is described in the sentences beginning with `After this, ...`.  What they don't explain is how the "retirement savings module" subtracts the exact amounts that were added when constructing the PUF records in the PUF-CPS data file.  If the amount added in and then subtracted out is not the same for a record, then the tax calculations are all fouled up.

What do you think about that reading of this paragraph?

**Comment by MattHJensen** [comment]
> @MattHJensen, Could this be saying that the earnings for PUF records in the PUF-CPS file already include tax-favored employee pension contributions and are, therefore, what I've been calling gross earnings?

That's certainly what it implies. But your and my understanding of PUF earnings implies the opposite. That's why I was confused.

**Comment by martinholmer** [comment]
@MattHJensen said:
> That's certainly what it implies. But your and my understanding of PUF earnings implies the opposite.

It seem likely that they have "fixed" the PUF earnings variable in their PUF-CPS input data file before any simulation begins.  Sort of like fixing the PUF earnings variables once in the taxdata repository instead of fixing them every time Tax-Calculator runs.  We do that sort of thing.

**Comment by martinholmer** [comment]
@MattHJensen,  There's been no discussion of issue #1549 (wrt lack of pension contribution information in Tax-Calculator input data sets) for over two weeks.

Seems like there is a real need to impute tax-favored pension contributions in both the PUF-based and CPS-based input data sets.

Where do you put this task in your list of priorities?

How does its priority compare with imputing itemized expense amounts for non-itemizers (something that has been pending for over a year)?

@andersonfrailey

**Comment by feenberg** [comment]
On Wed, 11 Oct 2017, Martin Holmer wrote:

> 
> @MattHJensen, There's been no discussion of issue #1549 (wrt lack of pension contribution information
> in Tax-Calculator input data sets) for over two weeks.
> 
> Seems like there is a real need to impute tax-favored pension contributions in both the PUF-based and
> CPS-based input data sets.
>

I think this could be done with the Survey of Consumer Finance file.

dan

**Comment by martinholmer** [comment]
@feenberg said:
>>  Seems like there is a real need to impute tax-favored pension contributions in both the PUF-based and CPS-based input data sets.
>
> I think this could be done with the Survey of Consumer Finance file.

Yes, I agree that would be the best choice.
That is what TPC does in its imputation of pension contribution amounts.

So, how is going to do this work?

**Comment by MattHJensen** [comment]
@martinholmer asked me:

> Where do you put this task in your list of priorities?
> 
> How does its priority compare with imputing itemized expense amounts for non-itemizers (something that has been pending for over a year)?

My view is that this is higher priority than imputing itemized expense amounts for non-itemizers but that it is lower priority than making the cps file with C-TAM imputations available on PolicyBrain or responding to the requests of our users during the tax reform debate. 

> So, [who] is going to do this work?

As important as this is, I don't foresee full-time AEI staff (myself, Anderson, Hank, Amy, Sean) having an opportunity to work on this until the new year and perhaps not until after March, although if someone feels as if s/he has the bandwidth, I could be proven wrong. I have added a help wanted tag and think it might make sense to address this shortcoming in the documentation.

**Comment by martinholmer** [comment]
@MattHJensen commented on the priority of imputing pension contributions in our input data sets in [this comment](https://github.com/open-source-economics/Tax-Calculator/issues/1549#issuecomment-336280881).

His views make perfect sense to me.

**Comment by martinholmer** [comment]
Tax-Calculator pull request #2056 adds imputed pension contributions to the `puf.csv` input file and extrapolates them to future years using the same growth factor that is used to extrapolate wages and salaries.  Now, when comparing Tax-Calculator wages and salaries to those projected by CBO (as was done in issue #1156), one would need to add the new `pencon_p` and `pencon_s` variables to the `e00200` variable (which is **net** earnings, not **gross** earnings).

It would be desirable to redo the comparison made in issue #1156 in the manner described above using the newest `puf.csv` file.  If the results of that comparison are still problematic, a new issue should be opened.  Meanwhile, the very old issues #1156 and #1549 are going to be closed.

@MattHJensen @feenberg @andersonfrailey @hdoupe

**Comment by MattHJensen** [comment]
Thanks @martinholmer!
